src/onemod/stage/base.py

```diff
@@ -303,23 +363,7 @@ def set_params(self, param_id: int) -> Any:
         for param_name, param_value in params.items():
             self.config[param_name] = param_value
 
-    @classmethod
-    def evaluate(
-        cls,
-        filepath: Path | str,
-        stage_name: str,
-        method: str = "run",
-        param_id: int | None = None,
-        *args,
-        **kwargs,
-    ) -> None:
-        """Evaluate stage method."""
-        stage = cls.from_json(filepath, stage_name)
-        if method in stage.skip_if:
-            raise ValueError(f"invalid method: {method}")
-        stage.__getattribute__(method)(param_id, *args, **kwargs)
-
-    def run(self, param_id: int) -> None:
+    def run(self, subset_id: int | None = None, param_id: int | None = None) -> None:
```

Comment:
`GroupedStage` doesn't use `param_id` and `CrossedStage` doesn't use `subset_id`, but since we are merging them into `ModelStage` this won't matter